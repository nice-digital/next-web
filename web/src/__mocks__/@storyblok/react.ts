export const storyblokInit = (): null => null;

export const apiPlugin = (): null => null;
