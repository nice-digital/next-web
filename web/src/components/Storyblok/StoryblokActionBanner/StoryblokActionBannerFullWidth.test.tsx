/* eslint-disable testing-library/no-node-access */
import { render } from "@testing-library/react";

import {
	StoryblokActionBannerFullWidth,
	type StoryblokActionBannerProps,
} from "./StoryblokActionBannerFullWidth";

const mockStoryblokActionBannerFullWidth: StoryblokActionBannerProps = {
	blok: {
		heading: "Mock action banner title",
		body: {
			type: "Mock rich text",
		},
		image: {
			id: 123456789,
			name: "Kittens mate.",
			filename: "https://placekitten.com/408/287",
		},
		cta: [
			{
				text: "Press me",
				link: {
					linktype: "url",
					url: "https://placekitten.com/408/287",
				},
				variant: "inverse",
				_uid: "asdfsadf",
				component: "buttonLink",
			},
		],
		_uid: "123456789",
		component: "actionBanner",
	},
};

const mockStoryblokActionBannerFullWidthSubtle: StoryblokActionBannerProps = {
	blok: {
		...mockStoryblokActionBannerFullWidth.blok,
		variant: "fullWidthSubtle",
	},
};

describe("Storyblok action banner full width component", () => {
	it("should match snapshot", () => {
		const { container } = render(
			<StoryblokActionBannerFullWidth {...mockStoryblokActionBannerFullWidth} />
		);
		expect(container).toMatchSnapshot();
	});

	it("should match snapshot for the subtle variant", () => {
		const { container } = render(
			<StoryblokActionBannerFullWidth
				{...mockStoryblokActionBannerFullWidthSubtle}
			/>
		);
		expect(container).toMatchSnapshot();
	});
});
