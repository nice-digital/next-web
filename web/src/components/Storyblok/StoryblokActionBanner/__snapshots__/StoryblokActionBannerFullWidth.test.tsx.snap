// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyblok action banner full width component should match snapshot 1`] = `
<div>
  <section
    class="action-banner action-banner--full-width"
    data-component="action-banner--fullWidth"
  >
    <div
      class="action-banner--full-width__image-container"
      style="background-image: url(https://placekitten.com/408/287/m/899x0/filters:quality%2880%29);"
    />
    <div
      class="action-banner__container"
    >
      <div
        class="action-banner__inner"
      >
        <div
          class="action-banner__text"
        >
          <h2
            class="action-banner__title"
          >
            Mock action banner title
          </h2>
          <div
            class="action-banner__intro"
          >
            <div
              class="storyblokRichTextWrapper"
              data-testid="storyblok-rich-text"
            />
          </div>
        </div>
        <div
          class="action-banner__actions"
        >
          <a
            class="btn btn--inverse"
            data-component="button--inverse"
            href="https://placekitten.com/408/287"
          >
            Press me
          </a>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`Storyblok action banner full width component should match snapshot for the subtle variant 1`] = `
<div>
  <section
    class="action-banner action-banner--full-width-subtle"
    data-component="action-banner--fullWidthSubtle"
  >
    <div
      class="action-banner--full-width-subtle__image-container"
      style="background-image: url(https://placekitten.com/408/287/m/899x0/filters:quality%2880%29);"
    />
    <div
      class="action-banner__container"
    >
      <div
        class="action-banner__inner"
      >
        <div
          class="action-banner__text"
        >
          <h2
            class="action-banner__title"
          >
            Mock action banner title
          </h2>
          <div
            class="action-banner__intro"
          >
            <div
              class="storyblokRichTextWrapper"
              data-testid="storyblok-rich-text"
            />
          </div>
        </div>
        <div
          class="action-banner__actions"
        >
          <a
            class="btn btn--inverse"
            data-component="button--inverse"
            href="https://placekitten.com/408/287"
          >
            Press me
          </a>
        </div>
      </div>
    </div>
  </section>
</div>
`;
